dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -174,6 +175,187 @@ def _common_inexact_type(default_dtype, *dtypes):
+            # to perform LU decomposition of a batch of general matrices
+            ht_lapack_ev, _ = li._getrf_batch(
+                a_sycl_queue,
+                a_h.get_array(),
+                ipiv_h.get_array(),
+                dev_info_h.get_array(),
+                n,
+                a_stride,
+                ipiv_stride,
+                batch_size,
+                [a_copy_ev],
+            )
+
+            ht_lapack_ev.wait()
+            a_ht_copy_ev.wait()
+
+            return (a_h, ipiv_h, dev_info_h)
+
+        else:
+            # Initialize lists for storing arrays and events for each batch
+            a_vecs = [None] * batch_size
+            ipiv_vecs = [None] * batch_size
+            dev_info_vecs = [None] * batch_size
+            a_ht_copy_ev = [None] * batch_size
+            ht_lapack_ev = [None] * batch_size
+
+            # Process each batch
+            for i in range(batch_size):
+                # Copy each 2D slice to a new array as getrf destroys the input matrix
+                a_vecs[i] = dpnp.empty_like(a[i], order="C", dtype=res_type)
+                (
+                    a_ht_copy_ev[i],
+                    a_copy_ev,
+                ) = ti._copy_usm_ndarray_into_usm_ndarray(
+                    src=a_usm_arr[i],
+                    dst=a_vecs[i].get_array(),
+                    sycl_queue=a_sycl_queue,
+                )
+                ipiv_vecs[i] = dpnp.empty(
+                    (n,),
+                    dtype=dpnp.int64,
+                    usm_type=a_usm_type,
+                    sycl_queue=a_sycl_queue,
+                )
+                dev_info_vecs[i] = dpnp.empty(
+                    (1,),
+                    dtype=dpnp.int64,
+                    usm_type=a_usm_type,
+                    sycl_queue=a_sycl_queue,
+                )
+
+                # Call the LAPACK extension function _getrf
+                # to perform LU decomposition on each batch in 'a_vecs[i]'
+                ht_lapack_ev[i], _ = li._getrf(
+                    a_sycl_queue,
+                    a_vecs[i].get_array(),
+                    ipiv_vecs[i].get_array(),
+                    dev_info_vecs[i].get_array(),
+                    n,
+                    [a_copy_ev],
+                )
+
+            for i in range(batch_size):
+                ht_lapack_ev[i].wait()
+                a_ht_copy_ev[i].wait()
+
+            # Reshape the results back to their original shape
+            out_v = dpnp.array(a_vecs, order="C").reshape(orig_shape)
+            out_ipiv = dpnp.array(ipiv_vecs).reshape(orig_shape[:-1])
+            out_dev_info = dpnp.array(dev_info_vecs).reshape(orig_shape[:-2])
+
+            return (out_v, out_ipiv, out_dev_info)
+
+    else:
+        a_usm_arr = dpnp.get_usm_ndarray(a)
+
+        # `a` must be copied because getrf destroys the input matrix
+        a_h = dpnp.empty_like(a, order="C", dtype=res_type)
+        ipiv_h = dpnp.empty(
```

Comment:
I'd propose to allocate memory for `ipiv` and `dev_info` after the memory copying of `a`